pyi.py

```diff
@@ -249,21 +243,38 @@ def handleNodeDelete(self, node: ast.AST) -> None:
 
 
 class PyiAwareFileChecker(checker.FileChecker):
-    def run_check(self, plugin: Plugin, **kwargs: Any) -> Any:
-        if self.filename == "-":
-            filename = self.options.stdin_display_name
-        else:
-            filename = self.filename
+    if FLAKE8_MAJOR_VERSION < 5:
```

Comment:
nit: could move the if statement into the function, since most of the logic is still the same

Reply:
Happy to do that, but I think it might increase the risk that we accidentally break compatibility with flake8 <5 in the future if we ever modify this function. (Should we test with both flake8 <5 and flake8 5+ in CI? How would we do that?)

Reply:
I'm fine with breaking flake8 support fairly soon (as in a few months). Therefore, as this is only temporary, I think either solution is fine.

Reply:
Cool, that sounds like a good plan. Dropping support for flake8 < 5.0.0 will be less of a maintenance burden for us, and a few months sounds like a reasonable cut-off period.

Since this is only temporary, I'll move it into the function, in that case.